src/crimson/net/SocketConnection.cc

```diff
@@ -79,24 +83,26 @@ void SocketConnection::read_tags_until_next_message()
           switch (buf[0]) {
           case CEPH_MSGR_TAG_MSG:
             // stop looping and notify read_header()
-            return seastar::make_ready_future<seastar::stop_iteration>(
-                seastar::stop_iteration::yes);
+            return seastar::make_ready_future<stop_t>(stop_t::yes);
           case CEPH_MSGR_TAG_ACK:
             return handle_ack();
           case CEPH_MSGR_TAG_KEEPALIVE:
             break;
           case CEPH_MSGR_TAG_KEEPALIVE2:
             return handle_keepalive2()
-              .then([this] { return seastar::stop_iteration::no; });
+              .then([this] {
+                  return seastar::make_ready_future<stop_t>(stop_t::no);
```

Comment:
no need to futurize the `stop_t`.

Reply:
I was thinking of returning a make_ready_future<stop_t> here will save us some efforts to be "registered to run later in the next iteration of the event loop" according to https://github.com/scylladb/seastar/blob/master/doc/tutorial.md#ready-futures . But I'm OK with either way.

Reply:
@cyx1231st i don't see how they are connected. on the contrary, wrapping the instant value to be returned right away using its futurized form introduces unnecessary overhead. i'd suggest read the implementation of `repeat()`.

Reply:
fixed.